### ISSUE TITLE
PSGetModuleInfo.xml invalidates the module .cat file

### ISSUE THREAD

**Issue by f0rt** [PowerShell/PSResourceGet#286]
Team,

After publishing and installing a module from PS Gallery we get one additional file called PSGetModuleInfo.xml. After calling Test-FileCatalog I get "ValidationFailed" result. This way our customers cannot verify the module content.

Thanks,
f012rt

**Comment by joeyaiello** [comment]
@anmenaga can we talk about this at some point? Got some more context, not sure how we should fix it.

**Comment by 4c74356b41** [comment]
@SydneyhSmith so tagging this with 3.0 resolved means this won't ever be fixed in 2.0 and we should not sign the modules?

**Comment by TravisEz13** [comment]
You can Authenticode sign individual files (where applicable, including psd1, psm1, ps1xml, and dlls), or NuGet sign the package.   PowerShellGet does not currently verify NuGet signing but that is how the package should be signed.

To enable verification a signature in PowerShell, you need to switch on AllSigned policy or use Windows Code Integrity.

@4c74356b1 Can you message me on twitter so I can understand your exact requirement?

**Comment by TravisEz13** [comment]
I created a script to verify the publisher signing and give detailed results to find what @4c74356b41's issue was.
 
https://gist.github.com/TravisEz13/70263b0bd6b1e747dac2d116245f419e
 
It gives this result for the module, which looks like this

![image](https://user-images.githubusercontent.com/10873629/97618216-01491b80-19dc-11eb-953d-095c7e57b977.png)

**Comment by kilasuit** [comment]
@4c74356b41 - You can 100% publish and install signed modules - what the issue is here is that the 'PSGetModuleInfo.xml' should be hidden on the filesystem, which it definitely is on windows 

> 
> 
> @ThomasNieto thanks for bringing this up, we are happy to explore if this fits into PowerShellGet 3.0....do you have any suggestions as to what you would like to see in PowerShellGet to fix this....one option would be to use the Save-PSResource cmdlet without the parameter `-IncludeXML` (which is currently available in 3.0), another way for users to check would be a -PublisherCheck parameter on Install (which has not yet been implemented)

@SydneyhSmith - re the above it would make sense to move to a single file for User installs & a single file for System wide installs as opposed to individual files dotted around in each installed module version folder? This would also make `Get-InstalledModule` quicker too

**Comment by 4c74356b41** [comment]
@TravisEz13 thats true, thats for figuring that out. funny enough, it started working after I removed those 2 files from the cat file. but the cat file still doesn't pass the check :)

**Comment by ThomasNieto** [comment]
@SydneyhSmith can we get this bug fix / enhancement worked on before v3 GA so there is no breaking change? The fix @kilasuit recommended https://github.com/PowerShell/PSResourceGet/issues/286#issuecomment-719848944 I feel is the best solution for this problem with added benefits to making discovery faster.

**Comment by SydneyhSmith** [comment]
@ThomasNieto happy to take another pass through this issue and see what we can do
